### ISSUE TITLE
archive/tar: write too long error when files are symlinked at least twice

### ISSUE THREAD

**Issue by calvinbui** [hashicorp/terraform#31895]
### Terraform Version

```shell
Terraform v1.3.1
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v3.68.0
+ provider registry.terraform.io/hashicorp/helm v2.4.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.7.1
```

### Terraform Configuration Files

```terraform
...terraform config...
```

### Debug Output

https://gist.github.com/calvinbui/044f82fb2fb6486f23750353bff6acdf

### Expected Behavior

`terraform apply` follows the symlink to the original file, as if it was in the same directory/workspace.

### Actual Behavior

Error: Failed to upload configuration files: failed copying file "/Users/calvin/common_file.tf" to archive: archive/tar: write too long

### Steps to Reproduce

I symlink files between different orgs/workspaces/environments if they share a common configuration. Sometimes files are double symlinked, like in the example below.

Create a folder structure like so and run `terraform init`, then `terraform apply` to run into the error.

```
.
├── application
│   ├── common
│   │   ├── bucket.tf
│   │   └── common_file.tf -> ../../common_file.tf
│   ├── workspace1
│   │   ├── bucket.tf -> ../common/bucket.tf
│   │   ├── common_file.tf -> ../common/common_file.tf
│   │   └── vars.tf
│   └── workspace2
│       ├── bucket.tf -> ../common/bucket.tf
│       ├── common_file.tf -> ../common/common_file.tf
│       └── vars.tf
└── common_file.tf
```

### Additional Context

This was working in Terraform <1.3.0.

I'm using tfenv on my Mac, but the same issue occurred on Linux without it.

**Comment by willhughes-au** [comment]
Confirmed this is also happening for me on linux: 

```
tf --version
Terraform v1.3.1
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v4.19.0
+ provider registry.terraform.io/hashicorp/helm v2.6.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.11.0
```

**Comment by sfroment** [comment]
same problem on macOS with v1.3.1 too

**Comment by calvinbui** [comment]
Issue still exists in 1.3.6

**Comment by brandonc** [comment]
hashicorp/go-tfe#623 was the prerequisite for a fix for this issue. That change will give us more options within terraform to pass additional options into the underlying config slug packer.

**Comment by sebasslash** [comment]
Hi! Following up here. The above change to go-slug will fix this issue and hopefully we can backport this dependency upgrade so that users aren't forced to upgrade their TF version. 

Reviews are more than welcome! ❤️

**Comment by calvinbui** [comment]
terraform 1.5.2 is using go-slug v0.11.1 which should've fixed this issue (it contains [@sebasslash's fix](https://github.com/hashicorp/go-slug/pull/37).

 but likewise it's still happening. should an issue be raised over in go-slug?

**Comment by sebasslash** [comment]
Really sorry for getting back to you late on this. Can you provide some more detail on your directory structure and which dir you're using as your working dir?

**Comment by calvinbui** [comment]
> Really sorry for getting back to you late on this. Can you provide some more detail on your directory structure and which dir you're using as your working dir?

https://github.com/calvinbui/terraform-symlink-example/tree/master/service-02

**Comment by sharccc** [comment]
HashiCorp internal reference IPL-4553

**Comment by sebasslash** [comment]
Thank you both for putting this together @calvinbui & @willhughes-au ❤️  -- We've created a ticket internally. I will investigate the bug this week & ideally slate a fix for 1.6.X.
